### ISSUE TITLE
[BUG] unrecognized arguments while following the instructions

### ISSUE THREAD

**Issue by Louis-d-C** [obynio/certbot-plugin-gandi#31]
I am currently trying to use this plugin on my debian 10 server to obtain a wildcard certificate. I followed every step, but when it came to the certbot command, it fails with the error 

`certbot: error: unrecognized arguments: --dns-gandi-credentials /etc/letsencrypt/gandi/gandi.ini`

**Comment by ferblape** [comment]
Try with the old syntax https://pypi.org/project/certbot-plugin-gandi/1.2.3/

**Comment by alessandro-hic** [comment]
Have you installed pip package on normal user and then run certobot on su ?

**Comment by Synchro** [comment]
Old syntax doesn't work either.

**Comment by hitokiri8x** [comment]
Have you installed the package as normal user and run certbot as SU?

**Comment by Synchro** [comment]
No, it's installed as root (using the EFF official snap) and run as root. Either way, I'm assuming I'm running into the problems described in #22. It's a shame that [Gandi's own plugin](https://github.com/Gandi/letsencrypt-gandi) is also defunct; they're not doing themselves any favours when [all their competitors are supported](https://snapcraft.io/search?q=certbot-dns)!

**Comment by hitokiri8x** [comment]
I've installed 2 weeks ago on my RPi, cannot recall the first command, but this is my working crontab command for renew.

`certbot renew -q --authenticator dns-gandi --dns-gandi-credentials /etc/letsencrypt/gandi.ini`

Certbot was installed though apt

**Comment by sblondon** [comment]
@Louis-d-C  Which package managers do you use to get this error (pip or debian)?

I have the same error with debian packages (Debian GNU/Linux 11 - bullseye)  but it works in a virtualenv and pip packages:

```
sudo apt install python3-venv
python3 -m venv venv
./venv/bin/pip install certbot
./venv/bin/pip install certbot-plugin-gandi
sudo ./venv/bin/certbot certonly --authenticator dns-gandi --dns-gandi-credentials /etc/letsencrypt/gandi.ini -d testdns.domain.tld
```

In case it helps, the versions are different:

| lib                            | debian  | pip      | 
| -------------------------- | ---------- | -------- |
| certbot                      | 1.12.0  | 1.22.0 |
| certbot-plugin-gandi | 1.2.5-3 | 1.3.2   |

**Comment by baradhili** [comment]
I found using `sudo certbot` and the syntax from https://pypi.org/project/certbot-plugin-gandi/1.2.3/ worked when I installed things from apt on ubuntu 20

**Comment by Korko** [comment]
Fixed on my side, the issue was that certbot was called with sudo while the plugin was install without sudo so only for my local user.

`sudo pip install certbot-plugin-gandi`
Did the trick

**Comment by HLFH** [comment]
@Louis-d-C Can you specify your version of certbot and certbot-plugin-gandi, and their installation mode (apt, snap, pip)?

@Synchro The snap installation mode is not supported. But this project is open to any PR that will add support to it. 

Debian supports the package `certbot-dns-gandi` with apt, but only the next versions Debian Bookworm (testing) and Debian Sid (unstable) support a decent version (1.4.2 at this time): https://packages.debian.org/search?keywords=python3-certbot-dns-gandi. And the installation via pip for `certbot` and `certbot-dns-gandi` should work without any issue.

@obynio We may have to close the issue as we did not get any reply from @Louis-d-C in one year.

**Comment by Louis-d-C** [comment]
@HLFH 

I gave up on the wildcard certificate and did every subdomain I needed manually because I could not find out how to make it work. As of now, I don't need an infinite number of certificates so I can't really spend more time trying to fix this issue.

The version of certbot was 1.20.0, certbot-plugin-gandi was version 1.3.2 and the os was debian 10.

**Comment by dvinz** [comment]
Hi,
Old syntax works for me.
debian 10.13
certbot              0.31.0
certbot-dns-ovh      0.31.0
certbot-nginx        0.31.0
certbot-plugin-gandi 1.2.5

```
root@toto:~#certbot certonly --authenticator dns-gandi --dns-gandi-credentials /etc/letsencrypt/gandi/gandi.ini -d domain.fr -d \*.domain.fr --server https://acme-v02.api.letsencrypt.org/directory
usage:
  certbot [SUBCOMMAND] [options] [-d DOMAIN] [-d DOMAIN] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
certificate.
certbot: error: unrecognized arguments: --dns-gandi-credentials /etc/letsencrypt/gandi/gandi.ini
```

With old syntax =

```
root@toto:~#certbot certonly -a certbot-plugin-gandi:dns --certbot-plugin-gandi:dns-credentials /etc/letsencrypt/gandi/gandi.ini -d domain.fr -d \*.domain.fr
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator certbot-plugin-gandi:dns, Installer None
Cert is due for renewal, auto-renewing...
Renewing an existing certificate
Performing the following challenges:
dns-01 challenge for domain.fr
dns-01 challenge for domain.fr
Waiting 10 seconds for DNS changes to propagate
Waiting for verification...
Cleaning up challenges

IMPORTANT NOTES:
 - Congratulations! Your certificate and chain have been saved
 ```

**Comment by Alexandero89** [comment]
> @Louis-d-C Which package managers do you use to get this error (pip or debian)?
> 
> I have the same error with debian packages (Debian GNU/Linux 11 - bullseye) but it works in a virtualenv and pip packages:
> 
> ```
> sudo apt install python3-venv
> python3 -m venv venv
> ./venv/bin/pip install certbot
> ./venv/bin/pip install certbot-plugin-gandi
> sudo ./venv/bin/certbot certonly --authenticator dns-gandi --dns-gandi-credentials /etc/letsencrypt/gandi.ini -d testdns.domain.tld
> ```
> 
> In case it helps, the versions are different:
> lib 	debian 	pip
> certbot 	1.12.0 	1.22.0
> certbot-plugin-gandi 	1.2.5-3 	1.3.2

This one fixed it for me.
Create its own venv and only install these two packages.
And then tell cron to use this venv.
